File: backend/utils/testing/factories/blogs.py
import factory
import factory.django
from apps.blogs.models import Blog, BlogPost
from factory.django import DjangoModelFactory
from faker import Faker
from utils.testing.factories.users import UserFactory

fake: Faker = Faker(["no-NO"])


class BlogFactory(DjangoModelFactory):
    class Meta:
        model = Blog

    name = fake.company()
    description = fake.paragraph(nb_sentences=10)


class BlogPostFactory(DjangoModelFactory):
    class Meta:
        model = BlogPost

    title = fake.company()
    text = fake.text()
    author = factory.SubFactory(UserFactory)

File: frontend/src/types/blogs.ts
import { User } from "../interfaces/users";
import { Organization } from "../interfaces/organizations";
import internal from "stream";

export type Blog = {
  blogPosts: BlogPost[];
};

export type BlogPost = {
  id: string;
  title: string;
  publishDate: string;
  text: string;
  author: User;
};
